cli/upgrade/upgrade.go

```diff
@@ -24,11 +24,39 @@ import (
 	"github.com/arduino/arduino-cli/cli/instance"
 	"github.com/arduino/arduino-cli/cli/output"
 	"github.com/arduino/arduino-cli/commands"
+	"github.com/arduino/arduino-cli/configuration"
 	rpc "github.com/arduino/arduino-cli/rpc/commands"
 	"github.com/sirupsen/logrus"
 	"github.com/spf13/cobra"
 )
 
+var postInstallFlags struct {
+	runPostInstall  bool
+	skipPostInstall bool
+}
+
+func detectSkipPostInstallValue() bool {
+	if postInstallFlags.runPostInstall && postInstallFlags.skipPostInstall {
+		feedback.Errorf("The flags --run-post-install and --skip-post-install can't be both set at the same time.")
+		os.Exit(errorcodes.ErrBadArgument)
+	}
+	if postInstallFlags.runPostInstall {
+		logrus.Info("Will run post-install by user request")
+		return false
+	}
+	if postInstallFlags.skipPostInstall {
+		logrus.Info("Will skip post-install by user request")
+		return true
+	}
+
+	if !configuration.IsInteractive {
+		logrus.Info("Not running from console, will skip post-install by default")
+		return true
+	}
+	logrus.Info("Running from console, will run post-install by default")
+	return false
+}
+
```

Comment:
This piece is duplicated from here: https://github.com/arduino/arduino-cli/blob/be39aa9e70e28715392d7d78b9e0d7ff757bd117/cli/core/install.go#L50-L80

Ideally, we should avoid duplicating because it's easy to forget to update both copies if we change something.
What about making the functions in `cli/core/install.go` public and use them? I mean the two functions:

`func addPostInstallFlagsToCommand(cmd *cobra.Command) {`
and
`func detectSkipPostInstallValue() bool {`

hopefully it doesn't create an import loop...